### PR TITLE
fix: convert repo to typescript

### PR DESCRIPTION
**Pull request by Danial-EP**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,5 +9,4 @@ package-lock.json
 *.rdb
 .env*
 localStorage
-schema.graphql
```
Comment (dazadep) [inline .gitignore:12]: we need this file Danial so that we can import the schema to postman and generate tests

**Review by dazadep — CHANGES_REQUESTED** [review]
we need schema.graphql file

File: src/types/index.js [hunk 2]
```diff
@@ -1,16 +0,0 @@
...
-const loadedFiles = loadFilesSync(__dirname, { extensions: ['graphql'], recursive: true });
-
-const typeDefs = mergeTypeDefs(loadedFiles);
-
-const printedTypeDefs = print(typeDefs);
-fs.writeFileSync('schema.graphql', printedTypeDefs);
-
-module.exports = {
-  typeDefs
-}
```
Comment (dazadep) [inline src/types/index.js:16]: also why can't we keep the logic for types in here?

File: .gitignore (hunk 1 above)
Comment (Danial-EP) [reply .gitignore]: Reverted.

File: src/types/index.js (hunk 2 above)
Comment (Danial-EP) [reply src/types/index.js:16]: It throws some error when I tried to do it in this file.

Comment (dazadep) [reply src/types/index.js:16]: what was the error?

Comment (Danial-EP) [reply src/types/index.js:16]:
```
/Users/danial.javanmardi/src/elasticpath-graphql-server/dist/schema/src/addResolversToSchema.js:18
    Object.keys(resolvers).forEach(typeName => {
                           ^
Error: "Product" defined in resolvers, but not in schema
    at /Users/danial.javanmardi/src/elasticpath-graphql-server/dist/schema/src/addResolversToSchema.js:35:23
    at Array.forEach (<anonymous>)
    at addResolversToSchema (/Users/danial.javanmardi/src/elasticpath-graphql-server/dist/schema/src/addResolversToSchema.js:18:28)
    at schemaTransforms (/Users/danial.javanmardi/src/elasticpath-graphql-server/dist/schema/src/makeExecutableSchema.js:66:41)
    at /Users/danial.javanmardi/src/elasticpath-graphql-server/dist/schema/src/makeExecutableSchema.js:108:65
    at Array.reduce (<anonymous>)
    at Object.makeExecutableSchema (/Users/danial.javanmardi/src/elasticpath-graphql-server/dist/schema/src/makeExecutableSchema.js:108:29)
    at Object.<anonymous> (/Users/danial.javanmardi/src/elasticpath-graphql-server/src/index.ts:32:16)
    at Module._compile (node:internal/modules/cjs/loader:1109:14)
    at Module.m._compile (/Users/danial.javanmardi/src/elasticpath-graphql-server/node_modules/ts-node/src/index.ts:1056:23)
[nodemon] app crashed - waiting for file changes before starting...
```

Comment (Danial-EP) [reply src/types/index.js:16]: I've fixed it, it was from the path. 

I did not want to spend time on it initially, and imho I prefer to have only `graphql` in this folder.

Comment (Danial-EP) [reply src/types/index.js:16]: I had to revert [this commit](https://github.com/elasticpath/elasticpath-graphql-server/pull/87/commits/2215ecf41748c591c12832b338350242eaafef77) as it failed in CI. Let me know how you feel about it, I can fix CI tomorrow if you have a strong opinion about it.

**Review by dazadep — APPROVED** [review]

**Pull request merged**